src/measure_theory/integral/circle_integral_transform.lean

```diff
@@ -0,0 +1,168 @@
+  (hf : continuous_on f (sphere z R)) (hw : w ∈ ball z R) :
+  continuous (circle_integral_transform_deriv R z w f) :=
+begin
+  rw circle_integral_transform_deriv_eq,
+  exact (circle_map_inv_continuous hR hw).smul (circle_integral_transform_continuous hR hf hw),
+end
+
+/--A useful bound for circle integrals (with complex codomain)-/
+def circle_integral_bounding_function (R : ℝ) (z : ℂ) : (ℂ × ℝ → ℂ) :=
+  λ w, circle_integral_transform_deriv R z w.1 (λ x, 1) w.2
+
+lemma circle_int_funct_cont_on_prod {R r : ℝ} (hR : 0 < R) (hr : r < R) {z : ℂ} :
+ continuous_on (λ (w : ℂ × ℝ), ((circle_map z R w.snd - w.fst)⁻¹) ^ 2)
+  ((closed_ball z r) ×ˢ (⊤ : set ℝ)) :=
+begin
+  simp_rw ←one_div,
+  apply_rules [continuous_on.pow, continuous_on.div, continuous_on_const],
+  refine ((continuous_circle_map z R).continuous_on.comp continuous_on_snd (λ _, and.right)).sub
+    (continuous_on_id.comp continuous_on_fst (λ _, and.left)),
+  simp only [mem_prod, mem_closed_ball, ne.def, and_imp, prod.forall],
+  intros a b ha hb,
+  apply circle_map_ne_on_ball hR,
+  simp only [mem_ball],
+  linarith,
+end
+
+lemma circle_integral_bounding_function_continuous {R r : ℝ} (hR : 0 < R) (hr : r < R) (z : ℂ) :
+  continuous_on (abs ∘ (circle_integral_bounding_function R z))
+  ((closed_ball z r) ×ˢ (⊤ : set ℝ) : set $ ℂ × ℝ) :=
+begin
+  have : continuous_on (circle_integral_bounding_function R z) (closed_ball z r ×ˢ (⊤ : set ℝ)),
+  { simp_rw [circle_integral_bounding_function],
+    apply_rules [continuous_on.smul, continuous_on_const],
+    simp only [deriv_circle_map],
+    have c := (continuous_circle_map 0 R).continuous_on,
+    apply_rules [continuous_on.mul, c.comp continuous_on_snd (λ _, and.right), continuous_on_const],
+    simp_rw ←inv_pow₀,
+    apply circle_int_funct_cont_on_prod hR hr, },
+  refine continuous_abs.continuous_on.comp this _,
+  show maps_to _ _ (⊤ : set ℂ),
+  simp [maps_to],
+end
+
+lemma circle_integral_bounding_function_bound {R r : ℝ} (hR: 0 < R) (hr : r < R) (hr' : 0 ≤ r)
+  (z : ℂ) :
+  ∃ (x : ((closed_ball z r) ×ˢ [0, 2 * π] : set $ ℂ × ℝ)),
+  ∀ (y : ((closed_ball z r) ×ˢ [0, 2 * π] : set $ ℂ × ℝ)),
+  abs (circle_integral_bounding_function R z y) ≤ abs (circle_integral_bounding_function R z x) :=
+begin
+  have cts := circle_integral_bounding_function_continuous hR hr z,
+  have comp : is_compact (((closed_ball z r) ×ˢ [0, 2 * π]) : set (ℂ × ℝ)),
+  { apply_rules [is_compact.prod, proper_space.is_compact_closed_ball z r, is_compact_interval], },
+  have none := (nonempty_closed_ball.2 hr').prod nonempty_interval,
+  simpa using is_compact.exists_forall_ge comp none (cts.mono (by { intro z, simp, tauto })),
+end
+
+/-- The derivative of a `circle_integral_transform` is bounded by a continuous function -/
+lemma circle_integral_transform_deriv_bound {R r : ℝ} (hR: 0 < R) (hr : r < R) (hr' : 0 ≤ r)
+  {z x : ℂ} {f : ℂ → ℂ} (hx : x ∈ ball z r) (hf : continuous_on f (sphere z R)) :
```

Comment:
I don't see the point of `r` here. Shouldn't we instead be proving:
```suggestion
lemma circle_integral_transform_deriv_bound {R : ℝ} (hR: 0 < R)
  {z x : ℂ} {f : ℂ → ℂ} (hx : x ∈ ball z R) (hf : continuous_on f (sphere z R)) :
```

Reply:
Looking at this again now I still think you should drop the `r` argument. I've just opened #14627 which may help.

In addition, isn't `circle_integral_transform_deriv R z y f t` periodic in `t`? So we should replace `∀ (t ∈ [0, 2 * π])` with `∀ (t : ℝ)`.

Reply:
Did you see my remark about replacing `∀ (t ∈ [0, 2 * π])` with `∀ (t : ℝ)`?

I think it's worth doing as it drops a hypothesis from the conclusion `∥circle_integral_transform_deriv R z y f t∥ ≤ B`.